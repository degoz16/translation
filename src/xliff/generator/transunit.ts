export type TransUnitParams = {
    target?: string;
    targetLangLocale?: string;
    source?: string;
    sourceLangLocale?: string;
    compact?: boolean;
    id: number;
};

export function transunit(parameters: TransUnitParams) {
    const {source, sourceLangLocale, target, targetLangLocale, compact} = parameters;

    let rendered = ``;

    if (target?.length) {
        rendered += '\n';
        rendered += `<target`;
        if (!compact) {
            rendered += ` xml:space="preserve"`;

            if (targetLangLocale?.length) {
                rendered += ` xml:lang="${targetLangLocale}"`;
            }
        }
        rendered += '>';
        rendered += `${target}</target>`;
    }

    if (source) {
        rendered += '\n';
        rendered += `<source`;
        if (!compact) {
            rendered += ` xml:space="preserve"`;

            if (sourceLangLocale?.length) {
                rendered += ` xml:lang="${sourceLangLocale}"`;
            }
        }
        rendered += '>';
        rendered += `${source}</source>`;
    }

    return rendered.trim();
}
