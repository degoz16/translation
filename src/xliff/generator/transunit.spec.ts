import {transunit} from './transunit';

describe('trans-unit generator', () => {
    it('generates trans-unit with provided: source, id', () => {
        const generated = transunit({source: 'Sentence about something', id: 1});

        expect(generated).toMatchSnapshot();
    });

    it('generates trans-unit with provided source, id, target', () => {
        const generated = transunit({
            source: 'Sentence about something',
            target: 'Предложение о чем-то',
            id: 1,
        });

        expect(generated).toMatchSnapshot();
    });
});
