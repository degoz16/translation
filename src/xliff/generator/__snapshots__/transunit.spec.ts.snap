// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`trans-unit generator generates trans-unit with provided source, id, target 1`] = `
"<target xml:space=\\"preserve\\">Предложение о чем-то</target>
<source xml:space=\\"preserve\\">Sentence about something</source>"
`;

exports[`trans-unit generator generates trans-unit with provided: source, id 1`] = `"<source xml:space=\\"preserve\\">Sentence about something</source>"`;
